src/server/snapshot.cc

```diff
@@ -253,24 +266,28 @@ bool SliceSnapshot::BucketSaveCb(PrimeIterator it) {
 }
 
 unsigned SliceSnapshot::SerializeBucket(DbIndex db_index, PrimeTable::bucket_iterator it) {
-  // Must be atomic because after after we call it.snapshot_version_ we're starting
+  // Must be atomic because after we call it.snapshot_version_ we're starting
   // to send incremental updates instead of serializing the whole bucket: We must not
   // send the update until the initial SerializeBucket is called.
   // Relying on the atomicity of SerializeBucket is Ok here because only one thread may handle this
   // bucket.
   FiberAtomicGuard fg;
   DCHECK_LT(it.GetVersion(), snapshot_version_);
 
+  bucket_ser_in_progress_ = true;
```

Comment:
this boolean is the same as serialize_bucket_running_

Reply:
yes yes, this was for testing, it's a draft PR I will clean this up -- I wanted to discuss it with Shahar